src/meta/src/controller/fragment.rs

```diff
@@ -1038,22 +1038,16 @@ impl CatalogController {
     pub async fn get_upstream_root_fragments(
         &self,
         upstream_job_ids: Vec<ObjectId>,
-        job_type: Option<PbTableJobType>,
     ) -> MetaResult<HashMap<ObjectId, PbFragment>> {
         let inner = self.inner.read().await;
 
         let mut fragments = Fragment::find()
             .filter(fragment::Column::JobId.is_in(upstream_job_ids))
             .all(&inner.db)
             .await?;
-        fragments.retain(|f| match job_type {
-            Some(PbTableJobType::SharedCdcSource) => {
-                f.fragment_type_mask & PbFragmentTypeFlag::Source as i32 != 0
-            }
-            // MV on MV, and other kinds of table job
```

Comment:
I think this isn't precise. See comment below

Reply:
I think if we want to do it strictly, we should pass `stream_job` and matches every case, instead of `table_job_type`. 

But I feel simply union all cases is good enough.